packages/integrations/preact/server.js

```diff
@@ -4,26 +4,74 @@ import StaticHtml from './static-html.js';
 
 const slotName = (str) => str.trim().replace(/[-_]([a-z])/g, (_, w) => w.toUpperCase());
 
+let originalConsoleError;
+let runningChecks = 0;
```

Comment:
This is going to be a nice quality of life improvement — thanks @hippotastic!

I’m curious about why the running total is needed. Could we uninstall the filter at the end of the check function instead? Looks like everything is synchronous, so I’d expect restoring `console.error = originalConsoleError` there to work, but there may be details I’m not seeing.

Reply:
The reason why I've added it this way it to make it safe for a future where `check` function calls may be run in parallel. You need some sort of reference counting in this case.

I didn't remove the hook afterwards because there's no clear moment when I can be sure that the function will not be called again and I wanted to avoid repeated removal and reinstallation.

Reply:
If the function is synchronous I don't think it could ever be meaningfully run in parallel? We would end up removing and reinstalling the handler a bunch but I don't know if that's a genuine performance issue.

Reply:
The `check` function of the `React` integration is async, so I figured that we can expect this function to be converted to an async one at one point as well, and I didn't want to leave unexpected surprises for future contributors there. Maybe I'm thinking too cautiously? I'd probably still want to leave this in here! 😄